File: doc/_i18n/en/tutorials/introduction/multi-robot-controller/p1.cpp
// In the header
#include <mc_tasks/SurfaceTransformTask.h>
// In the private members
std::shared_ptr<mc_tasks::SurfaceTransformTask> handTask;
// In the reset function
// Create the task and add it to the solver
handTask = std::make_shared<mc_tasks::SurfaceTransformTask>("RightGripper", robots(), 0, 5.0, 1000.0);
solver().addTask(handTask);
// Set a target relative to the handle position
handTask->target(sva::PTransformd(Eigen::Vector3d(0, 0, -0.025)) * robots().robot(1).surfacePose("Handle"));

File: doc/_i18n/en/tutorials/introduction/multi-robot-controller/p1.py
# In the reset callback
# Create the task and add it to the solver
self.handTask = mc_tasks.SurfaceTransformTask("RightGripper", self.robots(), 0, 5.0, 1000.0)
self.qpsolver.addTask(self.handTask)
# Set a target relative to the handle position
self.handTask.target(sva.PTransformd(eigen.Vector3d(0, 0, -0.025)) * self.robots().robot(1).surfacePose("Handle"))
